docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,953 @@
+
+```go
+  IsContractNativeMinterEnabled      bool
+  IsTxAllowListEnabled               bool
+  IsFeeConfigManagerEnabled          bool
+  IsHelloWorldEnabled                bool
+  // ADD YOUR PRECOMPILE HERE
+  // Is{YourPrecompile}Enabled       bool
+```
+
+We can add `IsHelloWorld()` which checks if we are equal or greater than the fork `blockTimestamp`. We use this to check out whether the precompile is enabled. We defined `GetHelloWorldConfig()` in the last step. The below code snippet can be copied and pasted to overwrite the default code.
+
+```go
+// IsHelloWorld returns whether [blockTimestamp] is either equal to the HelloWorld
+// fork block timestamp or greater.
+
+func (c *ChainConfig) IsHelloWorld(blockTimestamp *big.Int) bool {
+	config := c.GetHelloWorldConfig(blockTimestamp)
+	return config != nil && !config.Disable
+}
+```
+
+We can now add it to the `AvalancheRules()` which creates and returns a new instance of `Rules`. The below code snippet can be copied and pasted to overwrite the default code.
+
+```go
+rules.IsContractNativeMinterEnabled = c.IsContractNativeMinter(blockTimestamp)
+rules.IsTxAllowListEnabled = c.IsTxAllowList(blockTimestamp)
+rules.IsFeeConfigManagerEnabled = c.IsFeeConfigManager(blockTimestamp)
+rules.IsHelloWorldEnabled = c.IsHelloWorld(blockTimestamp)
+// ADD YOUR PRECOMPILE HERE
+// rules.Is{YourPrecompile}Enabled = c.{IsYourPrecompile}(blockTimestamp)
+```
+
+Done! All we had to do was follow the comments.
+
+### Step 6: Add Test Contract
+
+Let's add our test contract to `./contract-examples/contracts`. This smart contract lets us interact with our precompile! We cast the HelloWorld precompile address to the IHelloWorld interface. In doing so, `helloWorld` is now a contract of type `IHelloWorld` and when we call any functions on that contract, we will be redirected to the HelloWorld precompile address. The below code snippet can be copied and pasted into a new file called `ExampleHelloWorld.sol`.
+
+```sol
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./IHelloWorld.sol";
+
+// ExampleHelloWorld shows how the HelloWorld precompile can be used in a smart contract.
+contract ExampleHelloWorld {
+  address constant HELLO_WORLD_ADDRESS = 0x0300000000000000000000000000000000000000;
+  IHelloWorld helloWorld = IHelloWorld(HELLO_WORLD_ADDRESS);
+
+  function getHello() public view returns (string memory) {
+    return helloWorld.sayHello();
+  }
+
+  function setGreeting(string calldata greeting) public {
+    helloWorld.setGreeting(greeting);
+  }
+}
+```
+
+Please note that this contract is simply a wrapper and is calling the precompile functions.
+
+### Step 7: Add Precompile Solidity Tests
+
+#### Step 7.1 Add Hardhat Test
```

Comment:
Please add : after `Step x.y" to make them consistent with others.

Reply:
thanks